ext/standard/pack.c

```diff
@@ -388,6 +388,12 @@ PHP_FUNCTION(pack)
 		switch ((int) code) {
 			case 'h':
 			case 'H':
+				if (arg == INT_MAX) {
+					efree(formatcodes);
+					efree(formatargs);
+					zend_value_error("Type %c: integer overflow in format string", code);
+					RETURN_THROWS();
+				}
 				INC_OUTPUTPOS((arg + (arg % 2)) / 2,1)	/* 4 bit per arg */
```

Comment:
Another approach may be to compute `arg/2 + arg%2` instead. I believe this does the same thing but can't overflow.

Reply:
ok that s more efficient.